### PR TITLE
Migrated from jshint -> eslint

### PR DESCRIPTION
**Pull request by amilajack**
Closes https://github.com/h5bp/html5-boilerplate/issues/1913

- [x] Adds eslint support
- [x] Removed all remaining jshint deps
- [x] Conformed all existing code styles to eslint style config
- [x] Used Airbnb config

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -149,18 +149,6 @@ gulp.task('copy:normalize', () =>
         .pipe(gulp.dest(`${dirs.dist}/css`))
 );
 
-gulp.task('lint:js', () =>
```
Comment (roblarsen) [inline gulpfile.babel.js]: This is removed but no corresponding task was added back in later.

File: gulpfile.babel.js [hunk 2]
```diff
@@ -175,7 +163,7 @@ gulp.task('archive', (done) => {
 
 gulp.task('build', (done) => {
     runSequence(
-        ['clean', 'lint:js'],
```
Comment (roblarsen) [inline gulpfile.babel.js]: As above

File: dist/js/plugins.js [hunk 3]
```diff
@@ -1,24 +1,23 @@
 // Avoid `console` errors in browsers that lack a console.
-(function() {
```
Comment (roblarsen) [inline dist/js/plugins.js]: Why touch this file at all?

File: src/js/plugins.js [hunk 4]
```diff
@@ -1,24 +1,23 @@
 // Avoid `console` errors in browsers that lack a console.
-(function() {
```
Comment (roblarsen) [inline src/js/plugins.js]: As above this seems like it should be a separate PR.

File: .eslintrc [hunk 5]
```diff
@@ -1,3 +1,12 @@
 {
-    "extends": "eslint:recommended"
```
Comment (roblarsen) [inline .eslintrc]: I would prefer to mirror the existing rules we have. I want to keep changes like this as tight as possible.  the choice of tool shouldn't force us to touch every JS file in the project to match an new and arbitrary set of rules.  
I want to discuss these rules. I, for example, don't like trailing commas. I'm open to discuss everything on the project, but changing a bunch of stuff at once diffuses the opportunity for discussion.

File: test/file_existence.js [hunk 6]
```diff
@@ -3,129 +3,117 @@ import fs from 'fs';
...
+  '.gitignore',
+  '.htaccess',
+  '404.html',
+  'apple-touch-icon.png',
+  'browserconfig.xml',
 
-    'css/', // for directories, a `/` character
+  'css/', // for directories, a `/` character
             // should be included at the end
-        'css/main.css',
```
Comment (roblarsen) [inline test/file_existence.js]: Keep this formatting. This mirrors the file system and helps folks follow the file existence pattern.

**Comment by roblarsen** [comment]
@amilajack I finaly had time to take a look at the changes and had some issues. There's also a merge conflict since we added modernizr 3.0.

**Comment by XhmikosR** [comment]
@roblarsen: you can always fix the issues and push to @amilajack's fork if they take a long time to respond :)

**Comment by roblarsen** [comment]
@XhmikosR it's been a while so I'm not actually expecting this to get updated. At some point I was going to revisit.

**Comment by roblarsen** [comment]
Closing this and we'll revisit in a more orderly fashion.
